src/Microsoft.Extensions.PlatformAbstractions/Native/PlatformApis.cs

```
@@ -0,0 +1,164 @@
+            //   VERSION_ID = "14.04"
+            //   HOME_URL = "http://www.ubuntu.com/"
+            //   SUPPORT_URL = "http://help.ubuntu.com/"
+            //   BUG_REPORT_URL = "http://bugs.launchpad.net/ubuntu/"
+            // We use ID and VERSION_ID
+
+            if (File.Exists("/etc/os-release"))
+            {
+                var lines = File.ReadAllLines("/etc/os-release");
+                var result = new DistroInfo();
+                foreach (var line in lines)
+                {
+                    if (line.StartsWith("ID=", StringComparison.Ordinal))
+                    {
+                        result.Id = line.Substring(3).Trim('"', '\'');
+                    }
+                    else if (line.StartsWith("VERSION_ID=", StringComparison.Ordinal))
+                    {
+                        result.VersionId = line.Substring(11).Trim('"', '\'');
+                    }
+                }
+                return result;
+            }
+            return null;
+        }
+
+        private static Platform DetermineOSPlatform()
+        {
+#if NET451
+            var platform = (int)Environment.OSVersion.Platform;
+            var isWindows = (platform != 4) && (platform != 6) && (platform != 128);
+
+            if (isWindows)
+            {
+                return Platform.Windows;
+            }
+            else
+            {
+                try
+                {
+                    var uname = NativeMethods.Unix.GetUname();
+                    if (string.Equals(uname, "Darwin", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return Platform.Darwin;
+                    }
+                    if (string.Equals(uname, "Linux", StringComparison.OrdinalIgnoreCase))
+                    {
+                        return Platform.Linux;
+                    }
+                    return Platform.Unknown;
+                }
+                catch
```

could just do an empty catch and ```return Platform.Unknown;``` after it so you don't have it twice.